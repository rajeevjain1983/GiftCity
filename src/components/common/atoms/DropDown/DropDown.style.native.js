// 9fbef606107a605d69c0edbcd8029e5d 
import styled, { css } from 'styled-components/native';

const getSecondaryVariation = props => {
  return `${
    props.dropDownStyle.lightBorder
      ? `border-bottom-width: ${props.dropDownStyle.border}px; border-bottom-color: ${
          props.theme.colors.FOOTER.DIVIDER
        }`
      : `border-bottom-width: ${props.dropDownStyle.border}px; border-bottom-color: ${
          props.theme.colors.BLACK
        }`
  }`;
};

const DropDownStyle = css`
  height: ${props => props.dropDownStyle.height}px;
  ${props =>
    props.variation === 'primary'
      ? `border: ${props.dropDownStyle.border}px solid ${props.theme.colors.BUTTON.WHITE.BORDER}`
      : getSecondaryVariation(props)};
  background-color: ${props =>
    props.variation === 'primary' || props.bgColor
      ? props.theme.colors.gray[500]
      : props.theme.colors.white};
  width: 100%;
`;

const HeaderContainer = styled.View`
  margin-right: ${props => props.theme.spacing.ELEM_SPACING.SM};
`;

const HeaderItemContainer = styled.View`
  width: 95%;
`;

const Row = styled.TouchableOpacity`
  display: flex;
  flex-direction: row;
  ${props =>
    props.variation === 'primary' ? 'justify-content: center' : 'justify-content:space-between'};
  align-items: center;
`;

const DropDownItemContainer = styled.TouchableHighlight.attrs({
  underlayColor: props => props.theme.colors.BUTTON.WHITE.ALT_FOCUS,
})`
  padding: ${props => props.theme.spacing.ELEM_SPACING.XXXS};
  background-color: ${props => props.theme.colors.WHITE};
  justify-content: ${props => (props.variation === 'primary' ? 'flex-start' : 'center')};
`;

const Separator = styled.View`
  background-color: ${props => props.theme.colors.BUTTON.WHITE.BORDER};
  height: 1px;
`;

const FlatList = styled.FlatList`
  flex: 1;
`;

const StyledLabel = styled.Text`
  font-size: ${props =>
    !props.isFocused
      ? props.theme.typography.fontSizes.fs14
      : props.theme.typography.fontSizes.fs10};
  color: ${props => props.theme.colors.BLACK};
  font-weight: ${props =>
    !props.isFocused
      ? props.theme.typography.fontWeights.regular
      : props.theme.typography.fontWeights.extrabold};
  font-family: ${props => props.theme.typography.fonts.secondary};
`;

const SelectedLabelView = styled.View`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  color: ${props => props.theme.colors.BLACK};
`;

const SHADOW = `
  shadow-opacity: 0.15;
  shadow-radius: 2px;
  shadow-color: ${props => props.theme.colors.black};
  shadow-offset: 0px 4px;
  elevation: 2;
`;

const FlatListWrapper = styled.View`
  ${SHADOW}
  width: ${props => props.width};
  height: ${props => props.height};
  padding-right: ${props => props.theme.spacing.ELEM_SPACING.XXXS};
`;

const getAdditionalStyleWrapper = props => {
  const { height, paddingTop } = props;
  return {
    ...(height && { height }),
    ...(paddingTop && { paddingTop }),
  };
};

const TouchableOpacityWrapper = styled.TouchableOpacity`
  ${getAdditionalStyleWrapper}
  position: relative;
`;

const getAdditionalStyleOverlayWrapper = props => {
  const { left, top, width } = props;
  return {
    ...(left && { left }),
    ...(top && { top }),
    ...(width && { width }),
  };
};

const OverLayView = styled.View`
  position: absolute;
  flex-direction: row;
  border: 1px solid ${props => props.theme.colors.BUTTON.WHITE.BORDER};
  border-top-width: 1px;
  margin-top: -1px;
  ${getAdditionalStyleOverlayWrapper}
`;

export {
  DropDownStyle,
  HeaderContainer,
  Row,
  OverLayView,
  DropDownItemContainer,
  Separator,
  FlatList,
  StyledLabel,
  SelectedLabelView,
  FlatListWrapper,
  HeaderItemContainer,
  TouchableOpacityWrapper,
};
