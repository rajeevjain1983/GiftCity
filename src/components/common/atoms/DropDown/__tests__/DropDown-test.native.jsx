// 9fbef606107a605d69c0edbcd8029e5d 
import React from 'react';
import { shallow } from 'enzyme';
import { DropDownVanilla } from '../views/DropDown.native';

describe('DropDown Test', () => {
  let component;

  const props = {
    data: [{ label: 'foo', value: 'foo' }],
    selectedValue: 'foo',
    onValueChange: jest.fn(),
    variation: 'primary',
    itemStyle: {
      color: 'foo',
    },
    heading: 'foo',
  };

  beforeEach(() => {
    component = shallow(<DropDownVanilla {...props} />);
  });

  it('should be defined', () => {
    expect(component).toBeDefined();
  });

  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render with dropdown open state', () => {
    component.setState({ dropDownIsOpen: true });
    expect(component).toMatchSnapshot();
  });

  it('test openDropDown with primary variation ', () => {
    props.variation = 'primary';
    component = shallow(<DropDownVanilla {...props} />);
    component.instance().openDropDown();
    expect(component.state('dropDownIsOpen')).toBe(true);
  });

  it('test closeDropDown', () => {
    const instance = component.instance();
    instance.closeDropDown();
    expect(component.state('dropDownIsOpen')).toBe(false);
  });

  it('test dropDownLayout', () => {
    const obj = {
      item: {
        label: 'foo',
      },
    };
    expect(component.instance().dropDownLayout(obj)).not.toBeNull();
  });

  it('test onDropDownItemClick', () => {
    const obj = {
      label: 'foo',
      value: 'foo',
    };
    component.instance().onDropDownItemClick(obj);
    expect(component.state('dropDownIsOpen')).toBe(false);
  });

  it('test flatlist keyExtractor', () => {
    const item = {
      key: 'foo',
    };
    component.setState({ dropDownIsOpen: true });
    const flatList = component.find('Styled(FlatList)');
    expect(flatList).toHaveLength(1);
    expect(flatList.props().keyExtractor(item)).toBe('foo');
  });

  it('test flatlist ItemSeparatorComponent', () => {
    component.setState({ dropDownIsOpen: true });
    const flatList = component.find('Styled(FlatList)');
    expect(flatList).toHaveLength(1);
    expect(flatList.props().ItemSeparatorComponent()).not.toBeNull();
  });

  it('test getDerivedStateFromProps', () => {
    const state = {
      selectedLabelState: 'abc',
    };
    const result = DropDownVanilla.getDerivedStateFromProps(props, state);
    expect(result).toEqual({ selectedLabelState: 'foo' });
  });

  it('test getDerivedStateFromProps with id as value', () => {
    const state = {
      selectedLabelState: 'abc',
    };
    props.data = [{ displayName: 'foo', id: 'foo' }];
    const result = DropDownVanilla.getDerivedStateFromProps(props, state);
    expect(result).toEqual({ selectedLabelState: 'foo' });
  });

  it('test setDropDownPosition func', () => {
    const position = { top: 100 };
    const windowHeight = 1087;
    const deviceRemainingHeight = 300;

    const instance = component.instance();

    instance.setDropDownPosition(position, deviceRemainingHeight, true, 200, windowHeight);
    expect(component.state('flatListHeight')).toEqual(100);

    instance.setDropDownPosition(position, deviceRemainingHeight, true, 400, windowHeight);
    expect(component.state('flatListHeight')).toEqual(200);

    instance.setDropDownPosition(position, deviceRemainingHeight, false, 1400, windowHeight);
    expect(component.state('flatListHeight')).toEqual(815.25);

    instance.setDropDownPosition(position, deviceRemainingHeight, false, 200, windowHeight);
    expect(component.state('flatListHeight')).toEqual(200);
  });
});
