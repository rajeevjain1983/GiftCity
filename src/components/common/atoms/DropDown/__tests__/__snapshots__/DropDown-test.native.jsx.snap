// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDown Test should render correctly 1`] = `
<View
  style={null}
>
  <Styled(Text)
    isFocused={true}
  >
    foo
  </Styled(Text)>
  <Styled(Component)
    arrowImageStyle={null}
    bounces={true}
    customDropDownHeight=""
    data={
      Array [
        Object {
          "label": "foo",
          "value": "foo",
        },
      ]
    }
    disabled={false}
    dropDownItemFontWeight="semibold"
    dropDownStyle={null}
    expanded={false}
    heading="foo"
    highlightStyle=""
    isAnimateList={true}
    itemStyle={
      Object {
        "color": "foo",
      }
    }
    onPress={[Function]}
    onPressOut={[Function]}
    onValueChange={[MockFunction]}
    openDropdownOnLoad={false}
    pointerEvents="auto"
    selectedItemFontWeight="semibold"
    selectedValue="foo"
    variation="primary"
  >
    <Styled(View)>
      <Styled(BodyCopy)
        accessibilityText=""
        color="gray.800"
        dataLocator=""
        fontFamily="secondary"
        fontSize="fs13"
        fontWeight="semibold"
        margin={null}
        numberOfLines={null}
        text="foo"
        textAlign="center"
        textDecoration={null}
      />
    </Styled(View)>
    <Image
      accessibilityElementsHidden={true}
      alt=""
      importantForAccessibility="no-hide-descendants"
      source={
        Object {
          "testUri": "../../../src/assets/images/carrot-small-down.png",
        }
      }
      style={null}
    />
  </Styled(Component)>
  <Component
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  >
    <Styled(Component)
      accessibilityLabel="Tap to close it"
      accessibilityRole="none"
      accessible={true}
      activeOpacity={1}
      height={1334}
      onPress={[Function]}
      paddingTop={0}
    />
    <Styled(View)
      left={0}
      top={0}
      width={0}
    >
      <Styled(View)
        height={0}
        width={0}
      />
    </Styled(View)>
  </Component>
</View>
`;

exports[`DropDown Test should render with dropdown open state 1`] = `
<View
  style={null}
>
  <Styled(Text)
    isFocused={true}
  >
    foo
  </Styled(Text)>
  <Styled(Component)
    arrowImageStyle={null}
    bounces={true}
    customDropDownHeight=""
    data={
      Array [
        Object {
          "label": "foo",
          "value": "foo",
        },
      ]
    }
    disabled={false}
    dropDownItemFontWeight="semibold"
    dropDownStyle={null}
    expanded={true}
    heading="foo"
    highlightStyle=""
    isAnimateList={true}
    itemStyle={
      Object {
        "color": "foo",
      }
    }
    onPress={[Function]}
    onPressOut={[Function]}
    onValueChange={[MockFunction]}
    openDropdownOnLoad={false}
    pointerEvents="auto"
    selectedItemFontWeight="semibold"
    selectedValue="foo"
    variation="primary"
  >
    <Styled(View)>
      <Styled(BodyCopy)
        accessibilityText=""
        color="gray.800"
        dataLocator=""
        fontFamily="secondary"
        fontSize="fs13"
        fontWeight="semibold"
        margin={null}
        numberOfLines={null}
        text="foo"
        textAlign="center"
        textDecoration={null}
      />
    </Styled(View)>
    <Image
      accessibilityElementsHidden={true}
      alt=""
      importantForAccessibility="no-hide-descendants"
      source={
        Object {
          "testUri": "../../../src/assets/images/carrot-small-up.png",
        }
      }
      style={null}
    />
  </Styled(Component)>
  <Component
    hardwareAccelerated={false}
    transparent={true}
    visible={true}
  >
    <Styled(Component)
      accessibilityLabel="Tap to close it"
      accessibilityRole="none"
      accessible={true}
      activeOpacity={1}
      height={1334}
      onPress={[Function]}
      paddingTop={0}
    />
    <Styled(View)
      left={0}
      top={0}
      width={0}
    >
      <Styled(View)
        height={0}
        width={0}
      >
        <Styled(FlatList)
          ItemSeparatorComponent={[Function]}
          accessibilityRole="combobox"
          bounces={true}
          data={
            Array [
              Object {
                "label": "foo",
                "value": "foo",
              },
            ]
          }
          isFocused={true}
          keyExtractor={[Function]}
          renderItem={[Function]}
          style={
            Object {
              "height": 0,
            }
          }
        />
      </Styled(View)>
    </Styled(View)>
  </Component>
</View>
`;
